### PR TITLE
Improve type constraint for `\`

### PR DESCRIPTION
**Pull request by xukai92**

### CODE REVIEW

File: src/linalg.jl [hunk 1]
```diff
@@ -49,9 +49,7 @@ end
 
 # Matrix division
 
-const CuOrAdj = Union{CuVecOrMat, LinearAlgebra.Adjoint{T, CuVecOrMat{T}}, LinearAlgebra.Transpose{T, CuVecOrMat{T}}} where {T<:AbstractFloat}
-
-function Base.:\(_A::AT1, _B::AT2) where {AT1<:CuOrAdj, AT2<:CuOrAdj}
+function Base.:\(_A::CuMatOrAdj, _B::CuOrAdj)
```
Comment (xukai92) [inline src/linalg.jl]: A has to be a matrix while B could be a vector or matrix.

File: src/array.jl [hunk 2]
```diff
@@ -20,6 +20,9 @@ end
 CuVector{T} = CuArray{T,1}
 CuMatrix{T} = CuArray{T,2}
 CuVecOrMat{T} = Union{CuVector{T},CuMatrix{T}}
+CuMatOrAdj{T} = Union{CuMatrix, LinearAlgebra.Adjoint{T, CuMatrix{T}}, LinearAlgebra.Transpose{T, CuMatrix{T}}}
+CuOrAdj{T} = Union{CuVecOrMat, LinearAlgebra.Adjoint{T, CuVecOrMat{T}}, LinearAlgebra.Transpose{T, CuVecOrMat{T}}}
```
Comment (xukai92) [inline src/array.jl]: Are you happy with this?

**Comment by xukai92** [comment]
Actually fixed a bug for type constraint in https://github.com/JuliaGPU/CuArrays.jl/pull/460/commits/3eb88d5c31fe930c525e532c23d63fcb30e16648

File: src/array.jl (hunk 2 above)
Comment (maleadt) [reply src/array.jl]: Better add these type aliases to the linalg file

Comment (xukai92) [reply src/array.jl]: Moved.

**Comment by maleadt** [comment]
bors try

**Pull request merged**